include/odp/api/spec/packet.h

```diff
@@ -475,6 +475,22 @@ int odp_packet_reass_partial_state(odp_packet_t pkt, odp_packet_t frags[],
  */
 void *odp_packet_head(odp_packet_t pkt);
 
+/**
+ * Packet handle from packet head address
+ *
+ * Returns packet handle based on its head pointer value. Application must have ownership of
+ * the packet when requesting this conversion. Returned packet handle value (for the same pointer
+ * value) may change between packet ownership changes (e.g. due to handle generation counting).
+ * Application must not store and use handle values from previous ownerships of the packet.
```

Comment:
Thinking, more advanced use cases like VPP, avoid such limitations and since APP header is exposed to ODP spec in prev ODP API patch.
I think, it is better to have the following APIs instead of odp_packet_from_head()

1. Introduce new object similar to odp_packet_seg_t whose scope is get split and combine packet and have basic operation on the object like get/set data pointer and size etc LIKE
2. int odp_packet_app_seg_split(odp_packet_t pkt, odp_packet_app_seg_t *app_segs, int *num_segs);
3. odp_packet_t  odp_packet_app_seg_combine( odp_packet_app_seg_t *app_segs, int num_segs);
4. void* odp_packet_app_seg_header(odp_packet_app_seg_t  seg);
5.  odp_packet_app_seg_t  odp_packet_app_seg_from_header(odp_pool_t pool, void *app_header);
6. int odp_packet_app_seg_len(odp_packet_app_seg_t  seg);
7. void* odp_packet_app_seg_data(odp_packet_app_seg_t  seg);

Reply:
done in v8